### DISCUSSION TITLE
npm run bootstrap fails in coin-tester.

### DISCUSSION THREAD

**Discussion by ghost** [CoolBitX-Technology/coolwallet-js-sdk#363] [Q&A] [answered]
I'm trying to implement a new coin. I cloned the coolwallet-sdk repo. Inside it I'm trying to run coin-tester package. I've followed the instructions but executing ```npm run bootstrap``` fails with error ```coolwallet-sdk\packages\coin-tester\coin.config.json: Unexpected token � in JSON at position 0``` But the contents of my coin.config.json file are ```["@coolwallet/template"]```
This works in Linux Ubuntu but problem exist when using Windows. 
![Screenshot 2022-04-01 124951](https://user-images.githubusercontent.com/96065109/161214816-b8f76ba8-1cc1-45c7-be5f-af779e32ee9b.png)

**Comment by AKuzmichkin** [comment, marked as answer]
Create coin.config.json manually with such content:
[
    "@coolwallet/template"
]

**Reply by ghost** [reply]
Thanks. That worked, Now i get the prompt to choose from different options. But choosing any options results in 
```events.js:377
      throw er; // Unhandled 'error' event
      ^

Error: spawn npx ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:274:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:280:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npx',
  path: 'npx',
  spawnargs: [
    'lerna',
    'bootstrap',
    '--scope=@coolwallet/core',
    '--scope=@coolwallet/transport-jre-http',
    '--scope=@coolwallet/transport-web-ble'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tester@0.0.0 bootstrap: `babel-node -x ".tsx,.ts" tools/index.tsx`
npm ERR! Exit status 1```

**Reply by AKuzmichkin** [reply]
And this is not the last one (switched to Ubuntu)

**Reply by ghost** [reply]
I have ubuntu installed but my card isn't pairing with Bluetooth. Thats why i started trying with windows. I was thinking that the card wouldn't support ubuntu.

**Reply by AKuzmichkin** [reply]
Install Bluetooth Manager (Ubuntu) to pair.

**Reply by ghost** [reply]
Thanks. I will try that.

**Reply by ghost** [reply]
@AKuzmichkin Hey, How do you compose argument for signing. I can see that different coins are using different ways to generate the argument. Is there like a guideline or tutorial or something that would help me ?

**Reply by AKuzmichkin** [reply]
The payload (content of transaction buffer) for signing should be the same as if you sign it without cold wallet.

**Reply by ghost** [reply]
@AKuzmichkin If the transction object is MessagePack encoded before signing. Should the payload be Transaction Object or should it be MessagePack encoded form of the object. I don't see any functions related to MessagePack in script generator utils.
